connexion/security.py

```diff
@@ -403,6 +431,148 @@ async def wrapper(request, token, required_scopes):
 
         return wrapper
 
+
+class OIDCSecurityHandler(AbstractSecurityHandler):
+    """
+    SecurityHandlerFactory for the OIDC security scheme.
+    """
+
+    # TODO
+
+
+SECURITY_HANDLERS = {
+    # Swagger 2: `type: basic`
+    # OpenAPI 3: `type: http` and `scheme: basic`
+    "basic": BasicSecurityHandler,
+    # Swagger 2 and OpenAPI 3
+    "apiKey": ApiKeySecurityHandler,
+    "oauth2": OAuthSecurityHandler,
+    # OpenAPI 3 only
+    "openIdConnect": OIDCSecurityHandler,
+    # OpenAPI 3: http schemes
+    "bearer": BearerSecurityHandler,
+}
+
+
+# TODO: Split up
+#   along security schemes
+#   along 'functionality' -> get_*_func (resolver), security_{deny,passthrough}, and verify_*
+class SecurityHandlerFactory:
+    """
+    get_*_func -> _get_function -> get_function_from_name (name=security function defined in spec)
+        (if url defined instead of a function -> get_token_info_remote)
+
+    verify_* -> returns a security wrapper around the security function
+        check_* -> returns a function tasked with doing auth for use inside the verify wrapper
+            check helpers (used outside wrappers): _need_to_add_context_or_scopes
+            the security function
+
+        verify helpers (used inside wrappers): get_auth_header_value, get_cookie_value
+    """
+
+    no_value = AbstractSecurityHandler.no_value  # TODO
```

Comment:
Maybe better to define this on a module level since we're accessing it from different classes?